ScanToShop/ViewControllers/LoginViewController.m

```diff
@@ -19,12 +21,46 @@ @implementation LoginViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    self.passwordField.secureTextEntry = YES;
+}
+
+- (IBAction)onViewTap:(id)sender {
+    [self.view endEditing:YES];
 }
 
 - (IBAction)onSignInTap:(id)sender {
+    [self loginUser];
 }
 
 - (IBAction)onRegisterTap:(id)sender {
+    self.usernameField.text = @"";
+    self.passwordField.text = @"";
+}
+
+- (void)loginUser {
+    if ([self areUserInputFieldsEmpty]) {
+        [AlertManager loginAlert:@"" ViewController:self];
```

Comment:
Technically if the return type of this is void then you can return this but maybe that makes it less readable? Your call.